### ISSUE TITLE
[Question] Cyclomatic complexity calculation in test `main()` with null-aware operators.

### ISSUE THREAD

**Issue by solid-yuriiprykhodko** [dart-code-checker/dart-code-metrics#663]
I’m running into some high cyclomatic complexity numbers counted for my test file main() function.

The HTML report shows that it increments the complexity counter by 5 each time it sees a `.?` operation.

I’ve figured out that the +5 number comes from the nesting — I have 3 nested `group` statements that contain the test — see [example package](https://github.com/solid-yuriiprykhodko/metrics_cc_repro).

![image](https://user-images.githubusercontent.com/40931732/150988960-6b0a95d0-460a-4a3d-b5e8-db7ba97985e6.png)

Also, it adds up the +5 complexity for each `.?` statement inside the nested testcase.

Is this expected behaviour?
My expectation is that we would possibly have some complexity coming from nesting (the +5), but then only +1 from each .? statement.

If this is not the intended behaviour, perhaps we should consider excluding the test `main()` function declaration from the metric?

**Comment by dkrutskikh** [comment]
@solid-yuriiprykhodko good catch! Can you provide minimal project with reproduction of this issue?

**Comment by solid-yuriiprykhodko** [comment]
@dkrutskikh 
This repo serves as the minimal repro: https://github.com/solid-yuriiprykhodko/metrics_cc_repro.
Test `main` located here: https://github.com/solid-yuriiprykhodko/metrics_cc_repro/blob/master/test/test_metrics_test.dart#L5

**Comment by solid-yuriiprykhodko** [comment]
And the command that I use to run metrics:

```shell
dart run dart_code_metrics:metrics analyze lib test;
```

**Comment by dkrutskikh** [comment]
Available in 4.10.0 release 🚀
